gherkin/java/src/main/java/io/cucumber/gherkin/Token.java

```diff
@@ -10,6 +12,7 @@ class Token {
     public List<GherkinLineSpan> matchedItems;
     public int matchedIndent;
     public GherkinDialect matchedGherkinDialect;
+    public StepKeywordType keywordType;
```

Comment:
I think this would be better named `matchedKeywordType` as all the other fields that 'only exist after a match' are named that way. Also note https://github.com/cucumber/common/issues/1941

Reply:
this would make sense to be moved up next to the matchedKeyword property too

I do wonder if Keyword should be a small object at this stage (with the String+type), it might make things neater

Reply:
That's definitely an option (making Keyword a small object); I decided not to do that, because not all keywords are step keywords (Scenario/Rule/Example/...), but foremost because I was trying to keep the API changes (or the changes in general) as small as possible.

Reply:
Yep. Did the rename as you suggested.